src/Elliptic/DiscontinuousGalerkin/DgElementArray.hpp

```diff
@@ -58,24 +61,57 @@ struct DefaultElementsAllocator
     auto& local_cache = *Parallel::local_branch(global_cache);
     auto& element_array =
         Parallel::get_parallel_component<ParallelComponent>(local_cache);
+    const auto& initial_extents =
+        get<domain::Tags::InitialExtents<Dim>>(initialization_items);
+
     const auto& domain = Parallel::get<domain::Tags::Domain<Dim>>(local_cache);
     const auto& initial_refinement_levels =
         get<domain::Tags::InitialRefinementLevels<Dim>>(initialization_items);
-    const size_t num_of_procs_to_use =
-        static_cast<size_t>(sys::number_of_procs()) - procs_to_ignore.size();
+
+    const size_t number_of_procs =
+        Parallel::number_of_procs<size_t>(local_cache);
+    const size_t number_of_nodes =
+        Parallel::number_of_nodes<size_t>(local_cache);
+    const size_t num_of_procs_to_use = number_of_procs - procs_to_ignore.size();
+
     const domain::BlockZCurveProcDistribution<Dim> element_distribution{
         num_of_procs_to_use, initial_refinement_levels, procs_to_ignore};
+
+    // Will be used to print domain diagnostic info
+    std::vector<size_t> elements_per_core(number_of_procs, 0_st);
+    std::vector<size_t> elements_per_node(number_of_nodes, 0_st);
+    std::vector<size_t> grid_points_per_core(number_of_procs, 0_st);
+    std::vector<size_t> grid_points_per_node(number_of_nodes, 0_st);
+
     for (const auto& block : domain.blocks()) {
+      const auto& initial_extent = initial_extents[block.id()];
```

Comment:
Can do this in one line with `const size_t grid_points_per_element = alg::accumulate(initial_extents[block.id()], 1_st, std::multiplies<size_t>());`